frontend/src/components/charts/DurationSummaryChart.tsx

```diff
@@ -0,0 +1,100 @@
+import { useTheme } from "@mui/material/styles/";
+import { format } from "date-fns";
+import ReactEcharts from "echarts-for-react";
+
+import { Difficulty } from "~shared/types/base/index";
+
+type DurationSummaryProps = {
+  durationSummary:
+    | {
+        difficulty: Difficulty | string;
+        timetaken: number;
+        date: Date | string;
+      }[]
+    | undefined;
+};
+
+export const DurationSummaryChart = ({
+  durationSummary,
+}: DurationSummaryProps) => {
+  const theme = useTheme();
+  const getPastThirtyDays = () => {
+    const today = new Date();
+    const thirtyDaysAgo = new Date(new Date().setDate(today.getDate() - 29));
```

Comment:
Can use `subDays` instead.

Reply:
Will use subDays 30 instead of subDays 29 (Since statistics.service.ts uses 30)